### PR TITLE
Uppy for local file upload

### LINKED ISSUE #4378: Dropzone skips some files in multi-file uploads
When uploading a batch of files using Dropzone, the first couple of files will be loaded and the rest will often be skipped.  

Example: 
Screen Shot 2021-01-12 at 10 38 21 AM

The first two images will upload, and the rest will be skipped, disappearing from the Dropzone interface.

Possibly related - https://stackoverflow.com/questions/42421311/dropzonejs-is-randomly-skipping-files-to-send-to-the-server

### LINKED ISSUE #5752: Drag + Drop Ingest fails to ingest many files
### Summary

When I drag and drop a hundred files from my computer it uploads one or two and then refreshes and stops uploading.

This was reported by Christa Cleeton.

### Impact 

I can't upload a book from a connected hard drive on my computer.

### Priority recommendation

- [ ] asap
- [ ] within the next 3 weeks
- [x] PO will prioritize

## Sudden Priority Justification
_Required if "asap" or "within the next 3 weeks" is checked. Add "Sudden Priority" and "Maintenance/Research" labels_

### LINKED ISSUE #5796: Cannot attach raster xml file using dropzone
You can find an example raster xml metadata file in the spec fixtures.

There's some kind of file type filter, we think.

### PR DESCRIPTION
**Pull request by tpendragon**
This uses Uppy instead of vue-dropzone for local file upload, enabling resumable file uploads, large file uploads (gigabytes), and uploading of hundreds of files.

It leaves vue-dropzone in for now because it's in use for the OCR form, and I didn't want to refactor that just yet.

Screenshot 2023-05-17 at 4 34 58 PM
Screenshot 2023-05-17 at 4 35 06 PM
Screenshot 2023-05-17 at 4 35 15 PM
Screenshot 2023-05-17 at 4 35 21 PM

Depends on merge + deploy of https://github.com/pulibrary/princeton_ansible/pull/3913/files

This also replaces Jest with Vitest, because I couldn't get Jest to run.

Closes #4378
Closes #5796
Closes #5752

### CODE REVIEW

**Comment by hackartisan** [comment]
Does this PR address either https://github.com/pulibrary/figgy/issues/4378 or https://github.com/pulibrary/figgy/issues/5796? Or can it be linked to an issue explaining why we want to do this?

**Comment by tpendragon** [comment]
> Does this PR address either #4378 or #5796? Or can it be linked to an issue explaining why we want to do this?

Yup, should fix both.

**Comment by hackartisan** [comment]
> > Does this PR address either #4378 or #5796? Or can it be linked to an issue explaining why we want to do this?
> 
> This would be good to test to see if it addresses these issues, but should they hold up moving forward with Uppy? From the Uppy documentation, it seems like we can [specify what file types are allowed](https://uppy.io/docs/uppy/#restrictions) (see allowedFileTypes)

Yeah, my concern isn't that we are closing those tickets per se, but that the PR links to an issue that explains why we want to do the PR. Because someday someone will wonder, "why did we decide to do this?"

**Comment by tpendragon** [comment]
> Yeah, my concern isn't that we are closing those tickets per se, but that the PR links to an issue that explains why we want to do the PR. Because someday someone will wonder, "why did we decide to do this?"

:+1: One thing I was thinking about doing this was that we want to get video in some day too, so being able to support 100 GB over the web seems like a real thing.

File: app/javascript/test/setup.js [hunk 1]
```diff
@@ -2,9 +2,37 @@ import Vue from 'vue'
...
 Vue.component('v-details', VueDetails)
-jest.unmock('lodash')
-_.debounce = jest.fn((fn) => fn);
+vi.unmock('lodash')
+_.debounce = vi.fn((fn) => fn);
+expect.addSnapshotSerializer(vueSnapshotSerializer)
+
+const oldWindowLocation = window.location
+
+beforeAll(() => {
```
Comment (hackartisan) [inline app/javascript/test/setup.js:16]: Add a comment explaining what this is for

**Review by hackartisan — CHANGES_REQUESTED** [review]
This is awesome!

It might be nice to add documentation to the README about alternatives to yarn test for running vitest.

File: app/javascript/components/local_uploader.vue [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+      return document.getElementById(this.formId)
+    }
+  },
+  mounted () {
+    const uppy = new Uppy()
+      .use(DragDrop, {
+        target: '#drag-drop'
+      })
+      .use(StatusBar, { target: '#status-bar' })
+    uppy.use(Tus, { endpoint: TUS_ENDPOINT, limit: 6, chunkSize: (5 * 1024 * 1024) })
```
[the remark this answers is not in the document]
Comment (tpendragon) [reply app/javascript/components/local_uploader.vue:58]: Done.

File: app/javascript/test/setup.js (hunk 1 above)
Comment (tpendragon) [reply app/javascript/test/setup.js:19]: Done.

**Review by hackartisan — CHANGES_REQUESTED** [review]
@tpendragon I wouldn't want to squash merge this entirely but it does seem like it would be nice to squash into 2 commits: 1 that's mostly uppy and 1 that's mostly vitest. what do you think?

**Comment by tpendragon** [comment]
> @tpendragon I wouldn't want to squash merge this entirely but it does seem like it would be nice to squash into 2 commits: 1 that's mostly uppy and 1 that's mostly vitest. what do you think?

Sure, done.

**Review by hackartisan — APPROVED** [review]

**Pull request merged**